File: Marketplace.Core/DTO/MainCategoryDto.cs
namespace Marketplace.Core.DTO;

public class MainCategoryDto
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public string? PhotoUrl { get; set; }

    public IEnumerable<GetCategoryDto>? SubCategories { get; set; }
}

File: Marketplace.WebUI/marketplace-ui/src/app/dto/main-category.dto.ts
export class MainCategoryDto {
    route!: string;
    name!: string;
    photoUrl!: string;
    subCategories!: { route: string, name: string }[];
}

export class MainCategoryPostDto {
    name!: string;
    photoUrl!: string;
    subCategories!: CategoryDto[];
}

export class CategoryDto {
    id!: number;
    name!: string;
}